pyannote/audio/tasks/segmentation/mixins.py

```diff
@@ -422,6 +454,31 @@ def validation_step(self, batch, batch_idx: int):
             logger=True,
         )
 
+        # log validation segmentation loss
+        seg_target = None
+        seg_pred = None
+
+        if self.specifications.is_powerset_problem:
+            permutated_y, _ = permutate(y_pred, y)
+            seg_target = self.multilabel_to_powerset(permutated_y)
+            seg_pred = y_pred_powerset
+        elif self.specifications.problem == Problem.MULTI_LABEL_CLASSIFICATION:
+            permutated_y, _ = permutate(y_pred, y)
```

Comment:
This is specific to permutation-invariant tasks and should not go into this mixin.
Since, for now, Segmentation is the only permutation-invariant task, it makes sense to move this into `Segmentation.validation_step`.

For instance, it is wrong for VoiceActivityDetection or MultiLabelSegmentation task.
Therefore, it should be moved to Segmentation.